.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,64 @@
+name: Build and Test
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - synchronize
+jobs:
+  build-and-test:
+    strategy:
+      matrix:
+        ghc: ["8.0.2", "8.2.2", "8.4.4", "8.6.5", "8.8.3", "8.10.1"]
+        os: [ubuntu-latest]
+        include:
+          - os: macos-latest
+            ghc: "8.10.1"
+          - os: ubuntu-16.04
+            ghc: "7.10.3"
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Zsh
+        run: |
+          if [ "$RUNNER_OS" == "Linux" ]; then
+            sudo apt-get update
+            sudo apt-get install zsh
+          else
+            brew update
+            brew install zsh
+          fi
+      - uses: actions/setup-haskell@v1.1.4
+        id: setup-haskell-cabal
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-version: "3.0"
+      - run: cabal update --enable-tests
```

Comment:
@ibarrae I have more experience with `stack` but maybe @jpvillaisaza could provide more input here. AFAIK if you run `cabal configure --enable-tests` first then you don't need to pass this flag to all subsequent commands.

Reply:
Totally got me there :disappointed_relieved:, I'm not sure if that would be the case, I've seen some blog posts/CIs on other repos that use the same flags in all commands.

Reply:
@sestrella If you can `cabal configure` and add some options, that gets stored in a Cabal project file, but you don't need to run `cabal configure`, only if you don't want to specify options later on.

I would leave it with `--enable-tests`, but probably for the following one (`cabal build`), not this one. For this one (`cabal update`), I don't think `--enable-tests` does something, but I'm not sure.